### PR TITLE
Fix Tests MaxPool1D

### PR DESCRIPTION
**Pull request by Kranium2002**
Fix for #17043 

All tests passing. There was an issue with supported and unsupported datatypes.

### CODE REVIEW

**Review by khushi-411 — CHANGES_REQUESTED** [review]
The test failure looks relatable to me. Also, you probably might have to check the dtypes supported in the actual function, rather than explicitly writing dtypes in tests func.
```python
=================================== FAILURES ===================================
_____ test_maxpool1d_layer[cpu-ivy.functional.backends.numpy-False-False] ______
ivy_tests/test_ivy/test_stateful/test_layers.py:1215: in test_maxpool1d_layer
    method_tree="MaxPool1D.__call__",
ivy_tests/test_ivy/helpers/hypothesis_helpers/array_helpers.py:1731: in arrays_for_pooling
    dtype, x = draw(
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/hypothesis/internal/conjecture/data.py:951: in draw
    return strategy.do_draw(self)
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/hypothesis/strategies/_internal/lazy.py:156: in do_draw
    return data.draw(self.wrapped_strategy)
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/hypothesis/internal/conjecture/data.py:951: in draw
    return strategy.do_draw(self)
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/hypothesis/strategies/_internal/core.py:1558: in do_draw
    return self.definition(data.draw, *self.args, **self.kwargs)
ivy_tests/test_ivy/helpers/hypothesis_helpers/array_helpers.py:490: in dtype_and_values
    available_dtypes = draw(available_dtypes)
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/hypothesis/internal/conjecture/data.py:951: in draw
    return strategy.do_draw(self)
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/hypothesis/strategies/_internal/core.py:[155](https://github.com/unifyai/ivy/actions/runs/5312131480/jobs/9616234932?pr=17248#step:3:156)8: in do_draw
    return self.definition(data.draw, *self.args, **self.kwargs)
ivy_tests/test_ivy/helpers/hypothesis_helpers/dtype_helpers.py:173: in get_dtypes
    valid_dtypes = set(retrieval_fn(test_globals.CURRENT_BACKEND(), kind))
ivy_tests/test_ivy/helpers/hypothesis_helpers/dtype_helpers.py:35: in _get_fn_dtypes
    return test_globals.CURRENT_RUNNING_TEST.supported_device_dtypes[framework.backend][
E   KeyError: 'cpu'
E   
E   You can reproduce this example by temporarily adding @reproduce_failure('6.79.0', b'AAAAAA==') as a decorator on your test case
```

**Comment by Kranium2002** [comment]
Is there any way to tell the backend draw function to omit some data types from testing ?

**Comment by khushi-411** [comment]
> Is there any way to tell the backend draw function to omit some data types from testing ?

You could use the `@with_supported_dtypes` decorator in the main function file (not testing), or checkout as a parameter. Something like: https://github.com/unifyai/ivy/blob/a19462a76c46a5c884b243682c943733b52bf0b7/ivy_tests/test_ivy/test_stateful/test_initializers.py#L422

What's the result of applying the below decorator to the test function?
```python
@reproduce_failure('6.79.0', b'AAAAAA==')
```

**Comment by Kranium2002** [comment]
Hi, I just added with_supported_dtypes decorator and now all tests are passing on my local machine. 

How do we use the reproduce_failure decorator. I tried adding it above the test function but error keeps coming:

ERROR: not found: /workspaces/Ivy/ivy_tests/test_ivy/test_stateful/test_layers.py::test_maxpool1d_layer
(no name '/workspaces/Ivy/ivy_tests/test_ivy/test_stateful/test_layers.py::test_maxpool1d_layer' in any of [<Module test_layers.py>])

**Comment by Kranium2002** [comment]
Hi, is there anything else to be done in this PR?

File: ivy/functional/backends/jax/experimental/layers.py [hunk 1]
```diff
@@ -183,7 +183,7 @@ def general_pool(
         y = jnp.squeeze(y, axis=0)
     return y
 
-
+@with_supported_dtypes({'0.3.25 and below':('float64'),'0.3.25 and above':('float64')}, backend_version)
```
Comment (khushi-411) [inline ivy/functional/backends/jax/experimental/layers.py:186]: I see; there are two problems with this solution
- we still need two lines of spaces between two functions
- and the other when you are writing `{'0.3.25 and below':('float64'),'0.3.25 and above':('float64')}` along with the `@with_supported_dtypes`, you have already included float64 for all the versions. Why did you add it like that?

Thanks!

Comment (Kranium2002) [reply ivy/functional/backends/jax/experimental/layers.py:186]: I wanted to make sure that our code supports future versions of JAX and not only 0.3.25 and below which is why I wrote {'0.3.25 and below':('float64'),'0.3.25 and above':('float64')}

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
